src/codegen_call_user.cpp

```
@@ -0,0 +1,666 @@
+#include "ry/codegen.hpp"
+#include "ry/diagnostic.hpp"
+#include <climits>
+
+// ===== B4: emitUserFnCall =====
+
+llvm::Function *CodeGen::resolveOverload(const std::string &callee,
+                                          const std::vector<ExprPtr> &args,
+                                          std::vector<llvm::Value*> &outArgVals) {
+    auto fit = functions_.find(callee);
+    if (fit == functions_.end()) {
+        if (native_fn_sigs_.count(callee) || native_lib_index_.count(callee)) {
```

Clang (14.0.6)

[warning] if with identical then and else branches

(bugprone-branch-clone)